src/differential_geometry.jl

```diff
@@ -22,27 +72,133 @@ function Lift(X::Function, dependences::DataType...)::HamiltonianLift
 end
 
 # ---------------------------------------------------------------------------
-# Lie derivative of a scalar function along a vector field: L_X(f) = X⋅f
+# Lie derivative of a scalar function along a vector field or a function: L_X(f) = X⋅f
 
 # (postulat)
 # (X⋅f)(x)    = f'(x)⋅X(x)
 # (X⋅f)(t, x) = ∂ₓf(t, x)⋅X(t, x)
```

Comment:
@BaptisteCbl @ocots I maintain that this is not the correct definition for the non autonomous case. The action of the vector field should rather be defined according to
```math
X⋅f (t, x) = (\mathrm{d}/\mathrm{d}s)f(s, y(s, x))_{s = t}
```
where $y(s, x)$ is the solution at time t of $\dot{y}(s)=X(s, y(s))$, $y(s)=x$. That is
```math
X⋅f (t, x) =  \partial_t f(t, x) + \partial_x f(t, x) X(t, x).
```

Reply:
Before doing this we need to think globally, that is we have to consider also Poisson brackets. I don't remember what was confusing for me and why we have chosen to do this. I need to take a look to my notes.

Reply:
here was the remark: https://github.com/control-toolbox/CTBase.jl/issues/52#issuecomment-1568557356

Reply:
Actually, now we consider by default that the functions are autonomous, it may be more convenient to consider what you say, which was what we wanted at the first place.